components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/DefaultIDTokenBuilder.java

```diff
@@ -815,4 +816,29 @@ private String getIdpSessionKey(String accessToken) {
         }
         return idpSessionKey;
     }
+
+    /**
+     * Check the requirement of having a configured public certificate or JWKS in SP and throw an exception with an
+     * error message if the public certificate or JWKS in SP is not configured.
+     *
+     * @param clientId Client ID of the service provider.
+     * @param tenantDomain Tenant domain of the service provider.
+     * @throws IdentityOAuth2Exception
+     */
+    private void checkIfPublicCertConfigured(String clientId, String tenantDomain) throws IdentityOAuth2Exception {
+
+        try {
+            if (StringUtils.isBlank(OAuth2Util.getSPJwksUrl(clientId, tenantDomain))) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("Jwks uri is not configured for the service provider associated with " +
+                            "client_id: %s , Checking for x509 certificate.", clientId));
+                }
+                OAuth2Util.getX509CertOfOAuthApp(clientId, tenantDomain);
+            }
+        } catch (IdentityOAuth2Exception e) {
+            throw new IdentityOAuth2Exception("You have enabled ID token encryption without configuring a " +
```

Comment:
e is missing in the exception

Reply:
Addressed with 9cd2329bf504bbabc6cbac6f812b3a05182af131